src/data/date-formats.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Date separator characters
+ */
+type DateSeparator = '-' | '/' | ' ' | '. '
+
+/**
+ * Date formats supported by the parser and formatter
+ */
+export type Format =
+  | 'YYYY'
+  | `${'YYYY'}${DateSeparator}${'MM'}`
+  | `${'YYYY' | 'YYY' | 'YY'}${DateSeparator}${'MM'}${DateSeparator}${'DD'}`
+  | `${'DD'}${DateSeparator}${'MMMM' | 'MM'}${DateSeparator}${'YYYY'}`
+  | `${'MMMM'}${DateSeparator}${'DD'}${DateSeparator}${'YYYY'}`
+  | `${'MM'}${DateSeparator}${'DD'}`
+
+interface DateFormat {
+  regex: RegExp
+  format: Format // General format of the date
+}
```

Comment:
```suggestion
export type DateFormat =
  | 'YYYY'
  | `${'YYYY'}${DateSeparator}${'MM'}`
  | `${'YYYY' | 'YYY' | 'YY'}${DateSeparator}${'MM'}${DateSeparator}${'DD'}`
  | `${'DD'}${DateSeparator}${'MMMM' | 'MM'}${DateSeparator}${'YYYY'}`
  | `${'MMMM'}${DateSeparator}${'DD'}${DateSeparator}${'YYYY'}`
  | `${'MM'}${DateSeparator}${'DD'}`

interface DateFormatDetails {
  regex: RegExp
  format: Format // General format of the date
}
```

or some better naming maybe?
we have format method and format type so could be confusing?